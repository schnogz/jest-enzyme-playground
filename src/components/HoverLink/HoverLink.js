import React from 'react';
import styled from 'styled-components';

const Title = styled.a`
  font-size: 2em;
  color: palevioletred;
  text-decoration: none;
`;

export default class HoverLink extends React.Component {
	constructor(props) {
		super(props);

		this._onMouseEnter = this._onMouseEnter.bind(this);
		this._onMouseLeave = this._onMouseLeave.bind(this);

		this.state = {
			hovered: false,
		};
	}

	_onMouseEnter() {
		this.setState({hovered: true});
		console.info('hovered:', this.state.hovered);
	}

	_onMouseLeave() {
		this.setState({hovered: false});
		console.info('hovered:', this.state.hovered);
	}

	render() {
		return (
			<Title
				href={this.props.page || '#'}
				target={'_blank'}
				onMouseEnter={this._onMouseEnter}
				onMouseLeave={this._onMouseLeave}
			>
				{this.props.children}
			</Title>
		);
	}
}