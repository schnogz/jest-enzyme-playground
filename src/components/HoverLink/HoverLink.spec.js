import React from 'react';
import HoverLink from './HoverLink';
import { shallow } from 'enzyme'
import toJson from "enzyme-to-json";

describe('HoverLink component', () => {
	it('should render correctly', () => {
		const component = shallow(<HoverLink page="http://www.google.com">Google</HoverLink>);
		const tree = toJson(component);

		expect(tree).toMatchSnapshot();
	});

	it('should trigger mouse enter handler', () => {
		const component = shallow(<HoverLink page="http://www.google.com">Google</HoverLink>);
		const tree = toJson(component);

		// manually trigger the callback
		tree.props.onMouseEnter();

		expect(component.state().hovered).toBe(true);
	});

	it('should trigger mouse enter handler', () => {
		const component = shallow(<HoverLink page="http://www.google.com">Google</HoverLink>);
		const tree = toJson(component);

		// manually trigger the callback
		tree.props.onMouseEnter();
		tree.props.onMouseLeave();

		expect(component.state().hovered).toBe(false);
	});
});