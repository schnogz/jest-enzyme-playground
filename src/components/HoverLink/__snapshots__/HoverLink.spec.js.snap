// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HoverLink component should render correctly 1`] = `
<styled.a
  href="http://www.google.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  target="_blank"
>
  Google
</styled.a>
`;
